File: 3345-smallest-divisible-digit-product-i/3345-smallest-divisible-digit-product-i.py
class Solution(object):
    def smallestNumber(self, n, t):

        i = n

        while True:

            product = 1
            x = i

            while x:
                product *= x % 10
                x //= 10

            if product % t == 0:
                return i

            i += 1

File: 3345-smallest-divisible-digit-product-i/3345-smallest-divisible-digit-product-i.js
/**
 * @param {number} n
 * @param {number} t
 * @return {number}
 */
var smallestNumber = function(n, t) {

    for (let i = n;; i++) {

        let product = 1;

        for (let x = i; x > 0; x = Math.floor(x / 10))
            product *= x % 10;

        if (product % t === 0)
            return i;
    }
};